### ISSUE TITLE
[Improvement request] i18n scheme

### ISSUE THREAD

**Issue by fweng322** [pixelfed/pixelfed#3426]
So far I see the i18n scheme for pixelfed is to use php string replacement (under resource/lang/<locale>/*.php).  This way it is difficult to translate in certain sentence structure like:

in web.php,

'commented' => 'commented on your',

the variable before and after it would be in different order in Chinese, but with the current scheme we can not change the variable order.

I wish it can be changed.  AFAIK PHP also supports gettext which would provide a more flexible way for translators to precisely translate.

**Comment by dansup** [comment]
I agree, when I created the strings I did not realize it wouldn't work with other languages as I only speak english. I'm working on a solution to this!

**Comment by orangesunny** [comment]
I am slowly adopting Pixelfed, and I’m open to helping with some localization efforts personally.

Also, we at Weblate offer a solution that’s already boosting communities and making users happy for large numbers of projects: https://weblate.org/hosting/#libre. Let me know if I can help anyhow.

**Comment by fofwisdom** [comment]
#4497

**Comment by mrleemon** [comment]
I’d like to suggest considering the use of gettext for localization. gettext is a well-established, flexible, and efficient tool widely used for internationalizing software. By adopting it, the project could benefit from:

- Standardization: Aligning with a widely recognized standard in the industry.
- Ease of use: Simplified workflows for contributors familiar with translation tools like Poedit or similar.
- Scalability: Supporting projects with many languages and contributors.

**Comment by pacoorozco** [comment]
Laravel, the framework used to develop pixelfed, does not support gettext by default. There were some packages to provide gettext support to Laravel and most of them are now abandoned.

Instead, Laravel provides a good way to localize the applications and it's compliant with non-English variations where the attribute should be at the beginning or at the end. See it [here](https://laravel.com/docs/11.x/localization#replacing-parameters-in-translation-strings). 

Given the amount of work done to localize the app, I'd keep the current approach and just adopt the [parameters replacement in translation strings](https://laravel.com/docs/11.x/localization#replacing-parameters-in-translation-strings) where needed.

**WDYT?**

**Comment by keunes** [comment]
I [see](https://github.com/pixelfed/pixelfed/pull/5939) that the translations are stored in a JSON file - I assume that these parameters would also just appear in those files. Since Weblate already supports Jason files, so that shouldn't be an issue.

I just wonder if/how Weblate can write [pluralised strings the way Laravel expects them](https://laravel.com/docs/11.x/localization#pluralization).

**Comment by pacoorozco** [comment]
Laravel can use json filles, one per language, or multiple php files, which IMO are more suitable for this application. Weblate [is compatible with both](https://docs.weblate.org/es/latest/formats/laravel.html).

Current translations are using multiple PHP files for backend code and json/js files for the frontend side.

**Comment by shleeable** [comment]
Hello, the translation system has been refactored.

crowdin is now the source of truth for transcription.
